packages/api/src/Api.ts

```diff
@@ -72,3 +73,22 @@ export class Api extends ApiPromise {
         }
     }
 }
+
+export function getProvider(providerString: string): WsProvider | HttpProvider {
+    const URL_REGEX = /(wss?|https?|WSS?|HTTPS?):\/\/([-a-zA-Z0-9@:%._\+~#=]{2,256})\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/;
```

Comment:
use `/i` flag to ignore case https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/ignoreCase

Or better just use [URL](https://developer.mozilla.org/en/docs/Web/API/URL) to do the parsing

Because you already to lower case, so it won't be possible to have upper case in the match.

Reply:
Cool, thanks for the reviewing @xlc ! Right, I think it's better using the URL interface than handle this by myself. I will change it. But just not sure whether this interface is compatible with react native...